state-sync/storage-service/server/src/lib.rs

```diff
@@ -71,26 +73,76 @@ pub struct StorageServiceServer<T> {
     // TODO(philiphayes): would like a "multi-network" stream here, so we only
     // need one service for all networks.
     network_requests: StorageServiceNetworkEvents,
+    time_service: TimeService,
+
+    // We maintain a cached storage server summary to avoid hitting the DB for
+    // every request. This is refreshed periodically.
+    cached_storage_server_summary: Arc<RwLock<StorageServerSummary>>,
 }
 
 impl<T: StorageReaderInterface> StorageServiceServer<T> {
     pub fn new(
         config: StorageServiceConfig,
         executor: Handle,
         storage: T,
+        time_service: TimeService,
         network_requests: StorageServiceNetworkEvents,
     ) -> Self {
         let bounded_executor =
             BoundedExecutor::new(config.max_concurrent_requests as usize, executor);
+        let cached_storage_server_summary = Arc::new(RwLock::new(StorageServerSummary::default()));
+
         Self {
             config,
             bounded_executor,
             storage,
             network_requests,
+            time_service,
+            cached_storage_server_summary,
         }
     }
 
+    /// Spawns a non-terminating task that refreshes the cached storage server summary
+    async fn spawn_storage_summary_refresher(&mut self) {
+        let config = self.config;
+        let storage = self.storage.clone();
+        let time_service = self.time_service.clone();
+        let cached_storage_server_summary = self.cached_storage_server_summary.clone();
+
+        // Spawn the task
+        self.bounded_executor
+            .spawn(async move {
+                // Create a ticker for the refresh interval
+                let duration = Duration::from_millis(config.storage_summary_refresh_interval_ms);
+                let ticker = time_service.interval(duration);
+                futures::pin_mut!(ticker);
+
+                // Periodically refresh the cache
+                loop {
+                    ticker.next().await;
+
+                    if let Err(error) = refresh_cached_storage_summary(
+                        config,
+                        storage.clone(),
+                        cached_storage_server_summary.clone(),
+                    ) {
+                        let error = format!(
+                            "Failed to refresh the cached storage summary! Error: {:?}",
+                            error
+                        );
+                        error!(LogSchema::new(LogEntry::StorageServiceError).message(&error));
+                    }
+                }
+            })
+            .await;
+    }
```

Comment:
why not use the subscription service and get notified when a new commit happens and do it then?

Reply:
You could do that, but, the amount of wiring and refactoring it would take wouldn't be worth it right now. I feel like having a configurable refresh rate is a reasonable way to handle this as it stands today. :)